### PR TITLE
Added a script for converting NPMDB Overview into Markdown 

### PR DESCRIPTION
**Pull request by waveywaves**
This allows for the `overview` part of each of the entries in vuln/npm to have markdown parsable strings.
Related to #200

### CODE REVIEW

File: tools/migrations/transform_overview.js [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+//Pick up the overview from the file and parse it to return the changed string
+function parseOverview(file){
+    
+    if (file.overview != null){
+        var overview = JSON.stringify(file.overview);
+        overview = overview.replace(/\\n/g,'    ');
+        overview = overview.replace(/\\"/g,'"');
+        overview = overview.replace(/\\"/g,'"');
+    }
+    //console.log("###");
```
Comment (vdeturckheim) [inline tools/migrations/transform_overview.js]: Please remove debug logs

File: vuln/npm/1.json [hunk 2]
```diff
@@ -15,7 +15,7 @@
   ],
   "vulnerable_versions": "<=1.5.1",
   "patched_versions": ">=1.5.2",
-  "overview": "A vulnerability exists in bassmaster <= 1.5.1 that allows for an attacker to provide arbitrary JavaScript that is then executed server side via eval.",
+  "overview": "\"A vulnerability exists in bassmaster <= 1.5.1 that allows for an attacker to provide arbitrary JavaScript that is then executed server side via eval.\"",
```
Comment (vdeturckheim) [inline vuln/npm/1.json]: Why adding a `"` around them?

Comment (waveywaves) [reply vuln/npm/1.json]: I'll remove the unnecessary things and make the changes.

**Comment by waveywaves** [comment]
@vdeturckheim I have made the necessary changes. Do let me know if there needs to be any improvement.

**Comment by lirantal** [comment]
thanks @waveywaves for your contribution to the project ❤️ 👍 

I’m up for a more structured vulnerabilities database, though to put this PR in context - our `overview` field is taken from the hackerone report’s title field which means it is regarded as plain-text in its origin.

With that said, I know I've stated before the possibility of markdown but I don't think we reached consensus on that and we probably should discuss any check-list item there as a new issue to get engagement on, then announce the change with a complementing PR. This is to effectively provide any users of the vuln db enough time to prepare appropriately.

Would be happy to receive more input on this @vdeturckheim @pxlpnk @dgonzalez

**Comment by waveywaves** [comment]
@lirantal I feel that if we keep the overview as markdown it would be more helpful as we would be able to differentiate between what is code, what are the points and what is just plain text. It would provide some sort of semantics in terms of understanding as well as presenting the overview.
Considering github, which uses markdown in these comment boxes as well as in READMEs. Would moving over to plain text improve or degrade the quality of issue understandability ? I think that is what we are looking at here.

File: vuln/npm/102.json [hunk 3]
```diff
@@ -13,7 +13,7 @@
...
+  "overview": "sequalize is an Object-relational mapping, or a middleman to convert things from Postgres, MySQL, MariaDB, SQLite and Microsoft SQL Server into usable data for NodeJS        In Postgres, SQLite, and Microsoft SQL Server there is an issue where arrays are treated as strings and improperly escaped.        This causes potential SQL injection, where a malicious user could put `[test, ); DELETE TestTable WHERE Id = 1 --)]` inside of    ```    database.query(SELECT * FROM TestTable WHERE Name IN (:names), {      replacements: {        names: directCopyOfUserInput      }    });    ``` and cause the SQL statement to become `SELECT Id FROM Table WHERE Name IN (test, \\); DELETE TestTable WHERE Id = 1 --)`.         In Postgres, MSSQL, and SQLite, the backslash has no special meaning. This causes the the statement to delete whichever Id has a value of 1 in the TestTable table.",
```
Comment (jbreckman) [inline vuln/npm/102.json]: This code snippet no longer makes any sense without `'`

File: vuln/npm/117.json [hunk 4]
```diff
@@ -13,7 +13,7 @@
...
...ing if this might be possible vulnerability for many application which depends on shell-quote.        For example:           const quote = require(shell-quote).quote;       console.log(quote([foo>bar]));        will print foo>bar, where foo\\>bar is desirable.        This module is downloaded more than 1M times per month and many other modules are depending on this. If an application is escaping command-line args with this module, they might be vulnerable from malicious user input.        For example:    ```    var sq = require(../tests/get/shell-quote-1.6.0);    var exec = require(child_process).exec;        var pattern = process.argv[2];        command = sq.quote([grep, pattern]));    exec(cat file |  + command, function ( err, stdout, stderr) {        console.log(command, stdout);       });    ```    will be vulnerable when user input something like pattern = :</etc/passwd, which causes the content of /etc/passwd to be leaked.        Internally, (Jon Lamendola, Nick Starke, Jacob Waddell) found that the `;`, `{`, and `}` characters werent escaped properly either. This allows for full command injection. A malicious user could input `a;{echo,test,123,234}` to execute echo fully.",
```
Comment (jbreckman) [inline vuln/npm/117.json]: This code snippet also does not make sense without punctuation

**Comment by jbreckman** [comment]
Removing the punctuation actually makes some of the "overview" snippets very wrong/confusing.  There must be a better way.  Can't you escape punctuation if we are going to markdown?

Also, it's a little bit odd that 1 field out of 10 is "markdown".  Wouldn't it be safer/more correct to have all of the JSON fields be plain strings with no assumptions and to transform/escape them on the "edge".  In this case, the "edge" would likely be the script that pushes these json files to github, you can do the transform as part of that task.

**Comment by waveywaves** [comment]
@jbreckman Thanks for the insight :) Yeah that makes sense. Maybe it would be better if we can do a check based on the "parsability" of the overview. Maybe we could use this tool to do that https://github.com/remarkjs/remark-lint

**Comment by jbreckman** [comment]
@waveywaves Do you have samples of where this script helps markdown become "parseable"?  Markdown handles newlines quite well -- in fact it's needed for a number of markdown features like bullets, lists, etc.

**Comment by waveywaves** [comment]
@jbreckman I will work on the markdown parsability and should have something better soon .

**Comment by lirantal** [comment]
@waveywaves going to close this PR for now.
We can go back to re-opening it once we evaluate markdown support in the reports but for now not going to pursue it.

**Pull request closed without merging**